Hummingbird integration with lgbm converter

interesaaat opened the pull request:
Closes #403.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/onnx/onnxmltools?pullRequest=404) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/onnx/onnxmltools?pullRequest=404) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/onnx/onnxmltools?pullRequest=404) it.</sub>

interesaaat commented:
All tests pass. I don't know what is going on with LGTM, the logs look ok. @xadupre @wenbingl

fgsilva commented:
@interesaaat is there an update regarding this issue?

interesaaat commented:
@fgsilva the PR is ready, I am waiting for review.

interesaaat commented on .azure-pipelines/win32-conda-CI.yml:
```diff
@@ -68,6 +68,7 @@ jobs:
       echo Test numpy installation... && python -c "import numpy"
       python -m pip install %COREML_PATH% %ONNX_PATH%
       python -m pip install tensorflow-cpu==1.15.0
+      python -m pip install torch==1.5.0+cpu -f https://download.pytorch.org/whl/torch_stable.html
```
Agreed! Pytorch dependency is actually in Hummingbird requirements, however [we found](https://github.com/microsoft/hummingbird/issues/166#issue-644317532) that pytorch is not update on pip for windows. Another option is to install hummingbird directly from source rather than from pip. That would work. Should I install Hummingbird from github?

wenbingl replied:
In that case, can we put this line in the hummingbird requirements.txt : torch==1.5.0+cpu -f https://download.pytorch.org/whl/torch_stable.html, 
Or add some lines in the code to log this issue or avoid the confusion?

interesaaat replied:
This is complicated because `torch==1.5.0+cpu -f https://download.pytorch.org/whl/torch_stable.html` will not work for linux and mac. We have [this](https://github.com/microsoft/hummingbird/blob/40d8cc9477fc598081b973217e1315c9a0a890c7/setup.py#L25) logic to install PyTorch, but [we are having problems with pip from windows](https://github.com/microsoft/hummingbird/issues/166). I think that installing directly from github is probably the best solution right now.

interesaaat replied:
Ok installing from github does not work for python 3.7 because of [this](https://dev.azure.com/onnxmltools/onnxmltools/_build/results?buildId=13929&view=logs&j=14d29e63-7f60-55a7-fdd9-83f0bcb9c249&t=d805671f-b036-5223-a743-bcd6fb46af56). Basically `onnx` uses `typing` which for some reason creates problem with Hummingbird. Since you are already manually installing tensorflor, is it a problem if for windows we also manually install pytorch in the pipeline?

interesaaat commented on onnxmltools/convert/lightgbm/convert.py:
```diff
@@ -17,7 +18,7 @@
 def convert(model, name=None, initial_types=None, doc_string='', target_opset=None,
             targeted_onnx=onnx.__version__, custom_conversion_functions=None,
-            custom_shape_calculators=None):
+            custom_shape_calculators=None, onnx_operators_only=False):
```
Yes! Let me rename it.

wenbingl commented on onnxmltools/convert/common/utils.py:
```diff
@@ -5,3 +5,15 @@
 from onnxconverter_common.utils import *  # noqa
+
+
+def hummingbird_installed():
```
please move this func into https://github.com/microsoft/onnxconverter-common/blob/master/onnxconverter_common/utils.py.
 
this file will be deprecated.

interesaaat commented:
@wenbingl I think I addressed all of your comments. I have only modified the CI files to point to my fork of onnxconverter-common until the [PR](https://github.com/microsoft/onnxconverter-common/pull/134) is not in. I can remove those changes once merged.

wenbingl commented:
> @wenbingl I think I addressed all of your comments. I have only modified the CI files to point to my fork of onnxconverter-common until the [PR](https://github.com/microsoft/onnxconverter-common/pull/134) is not in. I can remove those changes once merged.

I merged the PR in onnxconverter-common so now you can update your final version and I will merge this PR then.

interesaaat commented:
Thanks! I am working on it right now. Will push in a minute.
